src/util/emitUtils.js

```diff
@@ -578,22 +580,28 @@ function emitToClients(io, nsp, rooms, key, obj) {
  */
 function doEmit(nsp, key, obj) {
   const newObjectAsString = getNewObjAsString(key, obj); // { key: {new: obj }}
-  const numClients = Object.values(nsp.connected).length;
+  const numClients = Object.values(nsp.rooms).length ?
+    Object.values(nsp.rooms).length : Object.values(nsp.connected).length;
```

Comment:
Good catch!
But this should be the total number of sockets in all rooms.
Probably makes sense to move this below within the if/else, so it can use sockets.length after the filter in the else (and Object.values(nsp.connected).length in the if)

Reply:
Cool, I've updated this. One thing I'm wondering though is if we should use sockets.length in both instances because this is the true number of clients being emitted to, whereas nsp.connected will be all of the sockets connected to the namespace.

Reply:
Oh yeah, you're right, we do need this for both. In that case, probably better to do it in one place above like you were doing before - I only suggested moving it into the with-callbacks block because I was thinking it was only needed there.

You could combine this with my above comment and set a sockets variable above, then use that for both the iteration and the numClients.

Reply:
Though I'm not sure I like having to filter all the sockets for every emit - could have a performance impact.
Maybe the call to trackEmit should only include numClients if the toggle is enabled?
Yeah, I think the only reason to track the number of clients emitted to is to have something to compare to the number of clients received.

So actually I think you should just remove this from the else and call trackEmit without the numClients arg. (and update trackEmit to allow this).

@iamigo thoughts?